tfe/resource_tfe_registry_module.go

```diff
@@ -24,18 +25,25 @@ func resourceTFERegistryModule() *schema.Resource {
 			"organization": {
 				Type:     schema.TypeString,
 				Computed: true,
+				Optional: true,
 			},
 			"module_provider": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:          schema.TypeString,
+				Computed:      true,
+				Optional:      true,
+				ForceNew:      true,
+				ExactlyOneOf:  []string{"vcs_repo"},
```

Comment:
Can you describe what this validation does? I couldn't find the docs

Reply:
In this validation I used both `ExactlyOneOf` and `ConflictsWith`. 

Even though both `module_provider` and `vcs_repo` attribute is optional, `ExactlyOneOf` will raise an error if none of them is specified.

`ConflictsWith` ensures that either a `module_provider`  or `vcs_repo` attribute is used but not `both`. Essentially they are mutually exclusive.

Now that I am rethinking it, seems like using only `ExactlyOneOf` might suffice. I will test that.